react/src/components/organisms/GenTeaser/utils.js

```diff
@@ -0,0 +1,96 @@
+  return`${new Date().getTime().toString()}_${n}`;
+};
+
+export const formatTime = (date) => {
+  const formattedDate = moment.utc(date).format('YYYYMMDDTHHmmssZ');
+  return formattedDate.replace('+00:00', 'Z');
+};
+
+export const calculateDuration = (startTime, endTime) => {
+  // snag parameters and format properly in UTC
+  const end = moment.utc(endTime).format('DD/MM/YYYY HH:mm:ss');
+  const start = moment.utc(startTime).format('DD/MM/YYYY HH:mm:ss');
+
+  // calculate the difference in milliseconds between the start and end times
+  const difference = moment(end, 'DD/MM/YYYY HH:mm:ss').diff(moment(start, 'DD/MM/YYYY HH:mm:ss'));
+
+  // convert difference from above to a proper momentJs duration object
+  const duration = moment.duration(difference);
+
+  return(
+    Math.floor(duration.asHours()) + moment.utc(difference).format(':mm')
+  );
+};
+
+export const buildUrl = (event, type, window) => {
+  let calendarUrl = '';
+
+  // allow mobile browsers to open the gmail data URI within native calendar app
+  // type = (type == "google" && this.isMobile()) ? "outlook" : type;
+
+  switch (type) {
+    case 'google':
+      calendarUrl = 'https://calendar.google.com/calendar/render';
+      calendarUrl += '?action=TEMPLATE';
+      calendarUrl += `&dates=${formatTime(event.startTime)}`;
+      calendarUrl += `/${formatTime(event.endTime)}`;
+      calendarUrl += `&location=${encodeURIComponent(event.location)}`;
+      calendarUrl += `&text=${encodeURIComponent(event.title)}`;
+      calendarUrl += `&details=${encodeURIComponent(event.description)}`;
+      break;
+
+    case 'yahoo': {
+      // yahoo doesn't utilize endTime so we need to calulate duration
+      const duration = calculateDuration(event.startTime, event.endTime);
+      calendarUrl = 'https://calendar.yahoo.com/?v=60&view=d&type=20';
+      calendarUrl += `&title=${encodeURIComponent(event.title)}`;
+      calendarUrl += `&st=${formatTime(event.startTime)}`;
+      calendarUrl += `&dur=${duration}`;
+      calendarUrl += `&desc=${encodeURIComponent(event.description)}`;
+      calendarUrl += `&in_loc=${encodeURIComponent(event.location)}`;
+      break;
+    }
+
+
+    case 'outlookcom':
+      calendarUrl = 'https://outlook.live.com/owa/?rru=addevent';
+      calendarUrl += `&startdt=${formatTime(event.startTime)}`;
+      calendarUrl += `&enddt=${formatTime(event.endTime)}`;
+      calendarUrl += `&subject=${encodeURIComponent(event.title)}`;
+      calendarUrl += `&location=${encodeURIComponent(event.location)}`;
+      calendarUrl += `&body=${encodeURIComponent(event.description)}`;
+      calendarUrl += '&allday=false';
+      calendarUrl += `&uid=${getRandomKey()}`;
+      calendarUrl += '&path=/calendar/view/Month';
+      break;
+
+    default:
+      calendarUrl = [
+        'BEGIN:VCALENDAR',
+        'VERSION:2.0',
+        'BEGIN:VEVENT',
+        `URL:${document.URL}`,
+        `DTSTART:${formatTime(event.startTime)}`,
```

Comment:
The startDate and endDate don't seem to reflect what's passed in in the downloaded file, seems to be adding today's date instead.

Reply:
Same with outlook and google cal

Reply:
![Screen Shot 2019-10-09 at 4 19 36 PM](https://user-images.githubusercontent.com/5789411/66517149-99fab980-eab0-11e9-989d-64baca28d42c.png)
<img width="538" alt="Screen Shot 2019-10-09 at 4 19 57 PM" src="https://user-images.githubusercontent.com/5789411/66517166-a3842180-eab0-11e9-9ee8-9f9ba744f080.png">

Reply:
@clairesunstudio --> fixed this just had the wrong variable called out in the utils function.

Reply:
confirmed fixed!